app/models/setting.rb

```diff
@@ -87,22 +86,19 @@ def self.humanized_category
     nil
   end
 
-  def self.cache_key(name)
-    "settings/#{name}"
-  end
-
   def self.[](name)
-    name = name.to_s
-    cache.fetch(cache_key(name)) do
-      find_by(:name => name)&.value
-    end
+    Foreman.setting_manager[name]
   end
 
   def self.[]=(name, value)
-    name   = name.to_s
-    record = where(:name => name).first!
-    record.value = value
-    record.save!
+    Foreman.setting_manager.update_setting(name, value)
+  end
+
+  def self.setting_type_from_value(default)
+    t = default.class.to_s.downcase
+    return t if TYPES.include?(t)
+    return "integer" if default.is_a?(Integer)
```

Comment:
I don't think that this condition is necessary because integer is taken by first one.

`5.class.to_s.downcase => "integer"`

And that brings me a question - How we can handle another numeric types ie. float or decimal?

Reply:
Not true, It can be and most likely will be `Fixnum`, that is sub-class of Integer, but `<fixnum>.class.to_s.downcase` won't help :)